lib/features/composer/data/repository/auto_complete_repository_impl.dart

```diff
@@ -1,18 +1,18 @@
 
 import 'package:core/core.dart';
 import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
-import 'package:tmail_ui_user/features/composer/data/datasource/composer_datasource.dart';
+import 'package:tmail_ui_user/features/composer/data/datasource/autocomplete_datasource.dart';
 import 'package:tmail_ui_user/features/composer/domain/model/auto_complete_pattern.dart';
 import 'package:tmail_ui_user/features/composer/domain/repository/auto_complete_repository.dart';
 
 class AutoCompleteRepositoryImpl extends AutoCompleteRepository {
 
-  final Map<DataSourceType, ComposerDataSource> composerDataSources;
+  final Map<DataSourceType, AutoCompleteDataSource> autoCompleteDataSources;
 
-  AutoCompleteRepositoryImpl(this.composerDataSources);
+  AutoCompleteRepositoryImpl(this.autoCompleteDataSources);
 
   @override
   Future<List<EmailAddress>> getAutoComplete(AutoCompletePattern autoCompletePattern) {
-    return composerDataSources[DataSourceType.local]!.getAutoComplete(autoCompletePattern);
+    return autoCompleteDataSources[DataSourceType.local]!.getAutoComplete(autoCompletePattern);
```

Comment:
`!` is quite dangerous, how about if `LocalAutoCompleteDataSourceImpl` is not bind

Reply:
Definitely must have. It belongs to local processing.

Reply:
hmm